### PR TITLE
HIPP-1318: Upgrade to Scala3

### PR DESCRIPTION
**Pull request by victorarbuesmallada**

### CODE REVIEW

File: app/uk/gov/hmrc/integrationcatalogue/repository/IntegrationRepository.scala [hunk 1]
```diff
@@ -299,7 +300,7 @@ class IntegrationRepository @Inject() (mongo: MongoComponent)(implicit ec: Execu
       FindOneAndUpdateOptions().upsert(false).returnDocument(ReturnDocument.AFTER)
     ).toFuture() map {
         case integrationDetail: IntegrationDetail => Some(integrationDetail)
-        case _ => None
+        case null => None
```
Comment (victorarbuesmallada) [inline app/uk/gov/hmrc/integrationcatalogue/repository/IntegrationRepository.scala:303]: a warning was raised recommending this approach

File: it/test/uk/gov/hmrc/integrationcatalogue/repository/IntegrationRepositoryISpec.scala [hunk 2]
```diff
@@ -47,8 +47,8 @@ class IntegrationRepositoryISpec
     with OasParsedItTestData
     with AwaitTestSupport {
 
-  lazy val repository: PlayMongoRepository[IntegrationDetail] = app.injector.instanceOf[IntegrationRepository]
```
Comment (victorarbuesmallada) [inline it/test/uk/gov/hmrc/integrationcatalogue/repository/IntegrationRepositoryISpec.scala:50]: cannot make lazy a non lazy overridden attribute

File: it/test/uk/gov/hmrc/integrationcatalogue/support/TestApplication.scala [hunk 3]
```diff
@@ -19,9 +19,7 @@ package uk.gov.hmrc.integrationcatalogue.support
 import play.api.Application
 import play.api.inject.guice.GuiceApplicationBuilder
 
-trait TestApplication {
-  _: BaseISpec =>
-
+trait TestApplication { this: BaseISpec =>
```
Comment (victorarbuesmallada) [inline it/test/uk/gov/hmrc/integrationcatalogue/support/TestApplication.scala:22]: Scala 3 self-types don't work with the `_` syntax
I went with what the documentation specifies:
https://docs.scala-lang.org/tour/self-types.html

File: project/AppDependencies.scala [hunk 4]
```diff
@@ -4,11 +4,11 @@ object AppDependencies {
 
   lazy val scalaCheckVersion = "1.14.0"
   lazy val enumeratumVersion = "1.8.0"
-  lazy val hmrcMongoVersion = "1.9.0"
-  lazy val bootstrapVersion = "8.6.0"
+  lazy val hmrcMongoVersion = "2.2.0"
+  lazy val bootstrapVersion = "9.0.0"
```
Comment (rdawson-hmrc) [inline project/AppDependencies.scala:8]: any reason not to go to the latest version 9.3.0? you can check internet library versions here: https://artefacts.tax.service.gov.uk/ui/packages?name=bootstrap-backend-play-30&type=packages

**Review by rdawson-hmrc — APPROVED** [review]

File: project/AppDependencies.scala (hunk 4 above)
Comment (victorarbuesmallada) [reply project/AppDependencies.scala:8]: good point, upgrading that now

**Review by rdawson-hmrc — APPROVED** [review]

**Review by rdawson-hmrc — APPROVED** [review]

**Pull request merged**